File: php/src/SharedKernel/Infrastructure/CommandBus/SimpleCommandBus.php
<?php

declare(strict_types=1);

namespace SharedKernel\Infrastructure\CommandBus;

use SharedKernel\Application\Command\Command;
use SharedKernel\Application\Command\Handler\Handler;
use SharedKernel\Application\CommandBus\CommandBus;

class SimpleCommandBus implements CommandBus
{
    /**
     * @var Handler[]
     */
    private array $handlers;

    public function registerHandler(Command $command, Handler $handler): void
    {
        $this->handlers[$command::class] = $handler;
    }

    public function handle(Command $command): void
    {
        $this->handlers[$command::class]->handle($command);
    }
}

File: Cs/SharedKernel/Infrastructure/CommandBus/SimpleCommandBus.cs
using Cs.SharedKernel.Application.Command;
using Cs.SharedKernel.Application.Command.Handler;
using Cs.SharedKernel.Application.CommandBus;
using System.Collections.Generic;

namespace Cs.SharedKernel.Infrastructure.CommandBus
{
    public class SimpleCommandBus : ICommandBus
    {
        private readonly Dictionary<string, IHandler> _handlers;

        public SimpleCommandBus()
        {
            _handlers = new Dictionary<string, IHandler>();
        }

        public void RegisterHandler(ICommand command, IHandler handler)
        {
            _handlers[command.GetType().Name] = handler;
        }

        public void Handle(ICommand command)
        {
            _handlers[command.GetType().Name].Handle(command);
        }
    }
}